api/src/main/java/org/pentaho/platform/api/genericfile/IGenericFileService.java

```diff
@@ -125,12 +131,38 @@ default boolean doesFileExist( @NonNull String path ) throws OperationFailedExce
    *
    * @param path The string representation of the path of the generic folder to create.
    * @return {@code true}, if the folder did not exist and was created; {@code false}, if the folder already existed.
-   * @throws AccessControlException   If the user of the current session does not have permission to create the folder.
-   * @throws InvalidPathException     If the folder's path is not valid.
-   * @throws OperationFailedException If the operation fails for any other (checked) reason.
+   * @throws AccessControlException    If the current user cannot perform this operation.
+   * @throws InvalidPathException      If the folder path is not valid.
+   * @throws InvalidOperationException If the path, or one of its prefixes, does not exist and cannot be created using this service (e.g. connections, buckets);
+   *                                   if the path or its longest existing prefix does not reference a folder;
+   *                                   if the path does not exist and the current user is not allowed to create folders on
+   *                                   the folder denoted by its longest existing prefix.
+   * @throws OperationFailedException  If the operation fails for some other (checked) reason.
+
    * @see #clearTreeCache()
    */
   default boolean createFolder( @Nullable String path ) throws OperationFailedException {
     return createFolder( GenericFilePath.parseRequired( path ) );
   }
+
+  /**
+   * Checks whether a generic file exists and the current user has the specified permissions on it.
+   * ...
```

Comment:
Missing doclet parameters (instead of the `...`).

Reply:
fixed